### PR TITLE
refactor from pyqtgraph to matplotlib

### LINKED ISSUE #213: refactor to matplotlib, as did mdaviz
The CI fail happens on `import pyqtgraph` and is not related to the code changes here (strictly Qt code). 

Following [mdaviz](https://github.com/BCDA-APS/mdaviz/issues/55#issuecomment-1930874388), considering a refactor to matplotlib anyway, at least for line charts.

_Originally posted by @prjemian in https://github.com/BCDA-APS/gemviz/issues/212#issuecomment-2010629820_

### PR DESCRIPTION
**Pull request by prjemian**
- close #213

### CODE REVIEW

**Comment by prjemian** [comment]
before (pyqtgraph) | after (matplotlib)
--- | ---
image | image

**Comment by prjemian** [comment]
still needs unit tests to be rectored

**Comment by prjemian** [comment]
Hrmph.  [`matplotlib.figure.Figure.get_suptitle()`](https://matplotlib.org/stable/api/figure_api.html#matplotlib.figure.Figure.get_suptitle) does not appear in the docs for [v3.7.3](https://matplotlib.org/3.7.3/api/figure_api.html) (used by the Py3.8 CI process).  Look for it with MPL v3.8+.

**Comment by prjemian** [comment]
Slight adjustment to plot title (will show the first and last dates of the data) and subtitle (catalog and date/time when plot was drawn).

image

**Comment by prjemian** [comment]
@rodolakis , @MDecarabas - Finally, ready for review.

**Comment by rodolakis** [comment]
I'm on it

File: gemviz/chartview.py [hunk 1]
```diff
@@ -34,10 +36,11 @@
     forestgreen
     salmon
 """.split()
-PLOT_SYMBOLS = """o + x star s d t t2 t3""".split()
+# https://matplotlib.org/stable/gallery/lines_bars_and_markers/marker_reference.html
+# from matplotlib.lines import Line2D
+# print(Line2D.markers)
```
Comment (MDecarabas) [inline gemviz/chartview.py:41]: Can be removed

File: gemviz/chartview.py [hunk 2]
```diff
@@ -54,14 +57,16 @@ def auto_symbol():
     return next(_AUTO_SYMBOL_CYCLE)
 
 
+# https://matplotlib.org/stable/users/index.html
+
+
```
Comment (MDecarabas) [inline gemviz/chartview.py:62]: Empty space

File: gemviz/chartview.py [hunk 3]
```diff
@@ -79,19 +84,24 @@ def __init__(self, parent, **kwargs):
...
+        # Adjust margins
+        self.figure.subplots_adjust(bottom=0.1, top=0.9, right=0.92)
+        # self.figure.tight_layout()
+        self.setOptions()
+
+        # https://stackoverflow.com/questions/16066695/add-an-extra-information-in-a-python-plot
+        # https://matplotlib.org/stable/gallery/text_labels_and_annotations/annotation_demo.html
+
+        # now = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        # self.figure.text(0, 0, f"plot: {now}", ha="left", size=8, color="lightgrey")
```
Comment (MDecarabas) [inline gemviz/chartview.py:100]: We don't need this merged into main

**Review by MDecarabas — COMMENTED** [review]
Small Changes but overall looks good

File: gemviz/chartview.py (hunk 2 above)
Comment (prjemian) [reply gemviz/chartview.py:62]: added by black

File: gemviz/chartview.py (hunk 3 above)
Comment (prjemian) [reply gemviz/chartview.py:100]: agreed.  I paid for line 100 with a few hours of search time.

File: gemviz/chartview.py (hunk 2 above)
Comment (prjemian) [reply gemviz/chartview.py:62]: Moving into module docstring.

File: gemviz/chartview.py (hunk 1 above)
Comment (prjemian) [reply gemviz/chartview.py:41]: Changing into an attribute docstring.

**Pull request merged**